base/weakkeydict.jl

```diff
@@ -1,22 +1,78 @@
 # weak key dictionaries
 
 type WeakKeyDict{K,V} <: Associative{K,V}
-    ht::Dict{Any,V}
+    ht::Dict{WeakRef,V}
```

Comment:
This change is unrelated.  I can revert it but it seems to me that the stricter type is a bit saver.